src/Features/LanguageServer/ProtocolUnitTests/Ordering/FailingMutatingRequestHandler.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.UnitTests.RequestOrdering
+{
+    [Shared, ExportLspMethod(MethodName, mutatesSolutionState: true)]
```

Comment:
We should export all of these test handlers with PartNotDiscoverable to ensure it won't automatically be included in places we don't want it.